stix_shifter_modules/secretserver/stix_transmission/api_client.py

```diff
@@ -0,0 +1,129 @@
+            respObj.message = "Unauthorized: Access token could not be generated."
+            #
+        return ResponseWrapper(respObj)
+
+    def get_status(self, search_id):
+        # It is a synchronous connector.
+        # return {"code": 200, "status": "COMPLETED"}
+        respObj = Response()
+        respObj.code = "200"
+        respObj.error_type = ""
+        respObj.status_code = 200
+        content = '{"search_id": "' + search_id + \
+                  '", "progress":100, "status":"COMPLETED", "data": {"message":"Completed for the search id provided."}}'
+        respObj._content = bytes(content, 'utf-8')
+        return ResponseWrapper(respObj)
+
+    def build_searchId(self):
+        # It should be called only ONCE when transmit query is called
+        # Structure of the search id is
+        # '{"query": ' + json.dumps(self.query) + ', "url" : ' + secretserverurl '}'
+        s_id = None
+        num = str(random.randint(0, 50))
+        if (self.query is None):
+            raise IOError(3001,
+                          "Could not generate search id because 'query' or 'authorization token' or 'credential info' is not available.")
+        else:
+            # id_str = '{"query": ' + json.dumps(self.query) + ', "target" : "' + self.url + '" + "random" : "' + str(random.randint(0,22)) + '"}'
+            id_str = '{"query": ' + json.dumps(
+                self.query) + ', "target" : "' + self.url + '", "random":"' + num + '"}'
+            id_byt = id_str.encode('utf-8')
+            s_id = base64.b64encode(id_byt).decode()
+            self.search_id = s_id
+        return s_id
+
+    def get_search_results(self, search_id, index_from=None, fetch_size=None):
+        # Sends a GET request from secret server
+        # This function calls secret server to get data
+        if (self.token):
+            self.search_id = search_id
+            date = self.decode_searchId()
+            if len(date) is not 0:
+                self.startDate = date[0]
+                self.endDate = date[1]
+            else:
+                lastupdate = (datetime.datetime.now() - datetime.timedelta(days=90)).replace(tzinfo=datetime.timezone.utc).timestamp()
+                self.startDate = datetime.datetime.date(datetime.fromtimestamp(lastupdate)).strftime("%s")
+                self.endDate =date.today()
+            resp = SecretServerApiClient.get_events(self)
+            return resp
+
+    def decode_searchId(self):
+        # These value (date, self.query) must be present.
+        try:
+            id_dec64 = base64.b64decode(self.search_id)
+            jObj = json.loads(id_dec64.decode('utf-8'))
+        except:
+            raise IOError(
+                3001, "Could not decode search id content - " + self.search_id)
+        self.query = jObj.get("query", None)
+        date = re.findall(r'\d{4}-\d{2}-\d{2}', self.query)
+        if date == []:
+           print("Please enter the date")
```

Comment:
this print statement doesn't make sense. you already have a default time range set for the search on line 105/6/7